src/meta/src/hummock/compaction/level_selector.rs

```diff
@@ -212,23 +219,63 @@ impl DynamicLevelSelector {
 }
 
 impl LevelSelector for DynamicLevelSelector {
-    fn need_compaction(&self, levels: &[Level], level_handlers: &mut [LevelHandler]) -> bool {
-        let ctx = self.get_priority_levels(levels, level_handlers);
-        ctx.score_levels
-            .first()
-            .map(|(score, _, _)| *score > SCORE_BASE)
-            .unwrap_or(false)
+    fn need_compaction(
+        &self,
+        levels: &[Level],
+        level_handlers: &mut [LevelHandler],
+        hash_mapping_manager: Option<&HashMappingManager>,
+    ) -> bool {
+        if hash_mapping_manager.is_some() {
+            true
+        } else {
+            let ctx = self.get_priority_levels(levels, level_handlers);
+            ctx.score_levels
+                .first()
+                .map(|(score, _, _)| *score > SCORE_BASE)
+                .unwrap_or(false)
+        }
     }
 
     fn pick_compaction(
         &self,
         task_id: HummockCompactionTaskId,
         levels: &[Level],
         level_handlers: &mut [LevelHandler],
+        hash_mapping_manager: Option<&HashMappingManager>,
     ) -> Option<SearchResult> {
         let ctx = self.get_priority_levels(levels, level_handlers);
         for (score, select_level, target_level) in ctx.score_levels {
             if score <= SCORE_BASE {
+                if let Some(hash_mapping_manager) = hash_mapping_manager {
```

Comment:
Would moving these code snippets to a new compaction picker keep level_selector simpler?